Sources/Layout/MessagesCollectionViewFlowLayout.swift

```diff
@@ -40,60 +49,25 @@ open class MessagesCollectionViewFlowLayout: UICollectionViewFlowLayout {
         }
     }
 
+    /// Font to be used by `TextMessageCell` for `MessageData.emoji(String)` case.
+    ///
+    /// The default value of this property is 2x the `messageLabelFont`.
+    internal var emojiLabelFont: UIFont
+
     /// Determines the maximum number of `MessageCollectionViewCell` attributes to cache.
     ///
     /// The default value of this property is 500.
     open var attributesCacheMaxSize: Int = 500
-    
-    /// A type property representing the `MessagesCollectionViewLayoutAttributes` to be used.
-    open override class var layoutAttributesClass: AnyClass {
-        return MessagesCollectionViewLayoutAttributes.self
-    }
-    
-    // MARK: - Properties [Private]
-    
-    /// Font to be used by `TextMessageCell` for `MessageData.emoji(String)` case.
-    ///
-    /// The default value of this property is 2x the `messageLabelFont`.
-    private var emojiLabelFont: UIFont
 
     typealias MessageID = String
     
     /// The cache for `MessageIntermediateLayoutAttributes`.
     /// The key is the `messageId` of the `MessageType`.
-    fileprivate var intermediateAttributesCache: [MessageID: MessageIntermediateLayoutAttributes] = [:]
-    
-    /// Convenience property for accessing the layout object's `MessagesCollectionView`.
-    fileprivate var messagesCollectionView: MessagesCollectionView {
-        guard let messagesCollectionView = collectionView as? MessagesCollectionView else {
-            fatalError(MessageKitError.layoutUsedOnForeignType)
-        }
-        return messagesCollectionView
-    }
+    fileprivate var layoutContextCache: [MessageID: MessageCellLayoutContext] = [:]
```

Comment:
I prefer to choose `NSCache` , it can control the number of object in memory.

Reply:
No `layoutContextCache` empty operation when `prepareLayout`?

Reply:
Do you think it's necessary? I don't even use the `prepareLayout` method 🤔

Reply:
@SD10 , I think it's necessary, dictionary has two reason, one is occupy memory, another is performance penalty , the more number of data, the more time consumption when get object because of hash conflict.

Yep, we don't use `prepareLayout`, I mean we should override it that prepare something like empty cache when layout needs recalculation.

Reply:
I currently empty the cache inside of here: https://github.com/MessageKit/MessageKit/blob/24932171d3cc319936d8c8f33bbf931299ebc014/Sources/Layout/MessagesCollectionViewFlowLayout.swift#L118

I'm fine with changing this to a more efficient cache 👍